internal/provider/namespaces_datasource.go

```diff
@@ -0,0 +1,445 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	cloudservicev1 "github.com/temporalio/terraform-provider-temporalcloud/proto/go/temporal/api/cloud/cloudservice/v1"
+	namespacev1 "github.com/temporalio/terraform-provider-temporalcloud/proto/go/temporal/api/cloud/namespace/v1"
+)
+
+var (
+	_ datasource.DataSource              = &namespacesDataSource{}
+	_ datasource.DataSourceWithConfigure = &namespacesDataSource{}
+)
+
+func NewNamespacesDataSource() datasource.DataSource {
+	return &namespacesDataSource{}
+}
+
+type (
+	namespacesDataSource struct {
+		client cloudservicev1.CloudServiceClient
+	}
+
+	namespacesDataModel struct {
+		ID         types.String         `tfsdk:"id"`
+		Namespaces []namespaceDataModel `tfsdk:"namespaces"`
```

Comment:
I'm looking at the design of similar data sources, like:

1. [aws_subnets](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets)
2. [aws_security_groups](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/security_groups)
3. [aws_route_tables](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/route_tables)

Each one of these has a schema that returns only the IDs of the resources that it finds, with the expectation that you'd pair this with a separate [`aws_subnet`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnet) data source later in the program. What do you think about doing something similar here? Something like `temporalcloud_namespaces` returning a list of namespace IDs, and `temporalcloud_namespace` (the datasource) turns an ID into the `namespaceDataModel` object you have here? I realize as I write this that `temporalcloud_regions` does the same thing as the data source you added here, so I'm not sure - as a user, which one of these schemes makes the most sense to you?

Reply:
I tend to favor the more verbose option (like how `temporalcloud_regions` does) since I often end up filtering by non-id attributes when doing bulk actions on resources. Just having the id means I end up in O(n) time instead of O(1) time (since I have to do n number of additional calls in those cases instead of one call to retrieve all the data I need) (and yeah I know with pagination we're not in O(n) anyway.) 

For my specific use case currently I _could_ get away with just having the id (or the ID and name, similar to how [the Datadog provider](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/data-sources/dashboard_list) often does it) but that will probably change and having all the attributes would be useful.

So especially since the Temporal Cloud API provides the full namespace spec, I heavily lean towards providing all the attrs rather than just id.

Reply:
works for me - thank you!